src/main/java/com/softserve/academy/spaced/repetition/service/impl/PointsTransactionServiceImpl.java

```diff
@@ -62,8 +68,8 @@ public void buyDeck(Long deckId) throws NotAuthorisedUserException, PointsTransa
             transaction.setCreationDate(new Date());
             transaction.setReference(deckOwnership.getReference());
             transactionRepository.save(transaction);
-            userService.updatePointsBalance(userFrom);
-            userService.updatePointsBalance(userTo);
+            publisher.publishEvent(new PointsBalanceEvent(this, userFrom));
```

Comment:
The same here. For example:
`this.getClass().getCanonicalName()`
to know the source of the evnt.

Reply:
Thanks for remarks. I made changes.